File: Java/src/main/java/com/gildedrose/AgedBrie.java
package com.gildedrose;

public class AgedBrie extends ItemStrategy {
    @Override
    public void update(Item item) {
        if (item.quality < 50) {
            incrementQuality(item);
        }
        decrementSellin(item);
        if (item.sellIn < 0) {
            if (item.quality < 50) {
                incrementQuality(item);
            }
        }
    }
}

File: csharp/AgedBrie.cs
namespace GildedRose
{
    public class AgedBrie : ItemStrategy
    {
        public override void Update(Item item)
        {
            IncrementQuality(item);

            DecrementSellin(item);

            if (item.SellIn < 0)
            {
                IncrementQuality(item);
            }
        }
    }
}